public/app/features/plugins/extensions/utils.tsx

```diff
@@ -110,3 +111,52 @@ export function generateExtensionId(pluginId: string, extensionConfig: PluginExt
     .reduce((s, c) => (Math.imul(31, s) + c.charCodeAt(0)) | 0, 0)
     .toString();
 }
+
+const _isProxy = Symbol('isReadOnlyProxy');
+
+/**
+ * Returns a proxy that wraps the given object in a way that makes it read only.
+ * If you try to modify the object a TypeError exception will be thrown.
+ *
+ * @param obj The object to make read only
+ * @returns A new read only object, does not modify the original object
+ */
+export function readOnlyProxy<T extends object>(obj: T): T {
+  if (!obj || typeof obj !== 'object' || isReadOnlyProxy(obj)) {
+    return obj;
+  }
+
+  const cache = new WeakMap();
+  const readonly = () => false;
```

Comment:
**Nit:** maybe I would "read this easier" if this function was inlined (but maybe that's just me):

(I know like this it wouldn't be that DRY.)

```typescript
defineProperty: () => false,
deleteProperty: () => false,
isExtensible: () => false,
set: () => false,
```